.github/workflows/ci.yml

```diff
@@ -113,9 +114,29 @@ jobs:
       - name: "Check out repository code"
         uses: "actions/checkout@v2"
       - name: "Setup environment"
-        uses: "networktocode/gh-action-setup-poetry-environment@v1"
+        uses: "networktocode/gh-action-setup-poetry-environment@v2"
+        with:
+          python-version: "${{ matrix.python-version }}"
       - name: "Copy credentials"
         run: "cp development/creds.example.env development/creds.env"
+      - name: "Remove Poetry Lock for Older Nautobot Tests"
+        run: "rm -f poetry.lock"
+        if: "matrix.nautobot-version != '1.3.0'"
+      - name: "Remove Black"
+        # We have dependency issues specifically with Nautobot 1.1 but we really don't care about black
+        # on Anything less than the latest
+        if: "matrix.nautobot-version != '1.3.0'"
+        run: "poetry remove black --dev"
+      - name: "Handle Markupsafe bug"
+        # Markupsafe 2.0.1 conflicts with the new jinja, but 2.1.x conflicts with Nautobot 1.1-2
+        # Markdown has issues with importlib in newer versions
+        if: "matrix.nautobot-version != '1.3.0'"
+        run: "poetry add markupsafe==2.0.1"
+      - name: "Set Nautobot install to the testing version"
+        run: "poetry add nautobot==${{ matrix.nautobot-version }}"
+      - name: "Remove Poetry Lock for Older Nautobot Tests"
+        run: "poetry update --lock"
```

Comment:
I don't really understand why this step is needed, but I believe you that it is. :-)

Reply:
I tend to agree with you but looking back at the GHA logs this step is actually doing stuff with the lock file so I'm hesitant to remove it...

Reply:
I am curious, instead of adding nautobot in the pyproject.toml, could we instead do it here?